components/dashboard/src/start/StartWorkspace.tsx

```diff
@@ -458,11 +467,20 @@ export default class StartWorkspace extends React.Component<StartWorkspaceProps,
     }
 
     redirectTo(url: string) {
+        if (this.state.redirected) {
+            console.info("Prevented another redirect", { url });
+            return;
+        }
         if (this.props.runsInIFrame) {
             this.ideFrontendService?.relocate(url);
         } else {
             window.location.href = url;
         }
+
+        this.setState({ redirected: true });
+        setTimeout(() => {
+            this.setState({ showRedirectMessage: true });
+        }, 2000);
```

Comment:
2s might be a tad too short, but let's see. Worked for me right now. :+1:

Reply:
Wanted to find a threshold which was not too short to prevent races, but also not too long to actually show up in time to be useful to the user. Will leave as is for now